dialogue-core/src/test/java/com/palantir/dialogue/core/RequestSizeMetricsChannelTest.java

```diff
@@ -0,0 +1,185 @@
+        TaggedMetricRegistry registry = DefaultTaggedMetricRegistry.getDefault();
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        int recordableRequestSize = 2 << 20;
+        byte[] expected = "a".repeat(recordableRequestSize).getBytes(StandardCharsets.UTF_8);
+        Request request = Request.builder()
+                .body(new RequestBody() {
+                    @Override
+                    public void writeTo(OutputStream output) throws IOException {
+                        output.write(expected);
+                    }
+
+                    @Override
+                    public String contentType() {
+                        return "text/plain";
+                    }
+
+                    @Override
+                    public boolean repeatable() {
+                        return true;
+                    }
+
+                    @Override
+                    public OptionalLong contentLength() {
+                        return OptionalLong.of(expected.length);
+                    }
+
+                    @Override
+                    public void close() {}
+                })
+                .build();
+
+        EndpointChannel channel = RequestSizeMetricsChannel.create(
+                ImmutableConfig.builder()
+                        .channelName("channelName")
+                        .channelFactory(factory)
+                        .rawConfig(ClientConfiguration.builder()
+                                .from(TestConfigurations.create("https://foo:10001"))
+                                .taggedMetricRegistry(registry)
+                                .build())
+                        .build(),
+                r -> {
+                    try {
+                        RequestBody body = r.body().get();
+                        body.writeTo(baos);
+                        body.close();
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                    return Futures.immediateFuture(new TestResponse().code(200));
+                },
+                TestEndpoint.GET);
+        ListenableFuture<Response> response = channel.execute(request);
+
+        assertThat(response.get().code()).isEqualTo(200);
+        Snapshot snapshot = DialogueClientMetrics.of(registry)
+                .requestsSize()
+                .channelName("channelName")
+                .serviceName("service")
+                .endpoint("endpoint")
+                .retryable("true")
+                .build()
+                .getSnapshot();
+        assertThat(snapshot.size()).isEqualTo(1);
+        assertThat(snapshot.get99thPercentile()).isEqualTo(expected.length);
+    }
+
+    @Test
+    public void small_request_not_recorded() throws ExecutionException, InterruptedException {
+        TaggedMetricRegistry registry = DefaultTaggedMetricRegistry.getDefault();
```

Comment:
Let's avoid the singleton instance, this way our test is less likely to be impacted by any other tests in the system:
```suggestion
        TaggedMetricRegistry registry = new DefaultTaggedMetricRegistry();
```